src/site/content/en/blog/godaddy/index.md

```diff
@@ -0,0 +1,97 @@
+
+### Focusing on our weakest link: Cumulative Layout Shift (CLS)
+
+Analyzing new data gave us a new perspective on what had to be done to improve website speed. Because of work done to improve the Lighthouse score, our 75th percentile LCP was 860 ms and our FID at the same threshold was below 10 ms, so we enjoyed a high pass rate for these metrics on our customer's sites: 78% and 98%, respectively. However, the [Cumulative Layout Shift (CLS)](/cls/) numbers look _quite different_ from what we were used to with Lighthouse. Our CLS at the 75th percentile was 0.17 - above the 0.1 threshold to "pass" - and our pass rate was thus only 47% over all of our sites.
+
+That metric dragged down our overall pass rate to 40%, so we decided to set an ambitious goal to move that number to above 60% by the end of August 2021. To do that, we would **have to focus on CLS**.
+
+In real life, users interact with the page and scroll past the "above the fold" content, which is something Core Web Vitals captures better. Due to the variability in how users interact with the site while it initially loads, CLS differed from lab and field data.
+
+## The road to passing all Core Web Vitals
+
+Improving performance takes trial and error, and every attempt doesn't always work as expected. However, here are a few improvements that helped us achieve our goals.
+
+Reserving space for loading images drastically improved our CLS score as it prevents the content below the images from shifting. We used the [CSS `aspect-ratio` property](https://developer.mozilla.org/docs/Web/CSS/aspect-ratio) to address this on those browsers which support it. For those that do not, we loaded a **transparent placeholder image** that was cached and just a few bytes in size, thus loading almost instantaneously.
+
+This generic image behavior allowed us to pre-calculate the final image height during server-side rendering, based on viewport width and image aspect ratio. This resulted in HTML markup with vertical space appropriately reserved for the final image. The improvement was particularly observable on mobile devices, since images are rendered to the full span of mobile viewports.
+
+Certain components on our customers sites have dynamic content (eg. list of external customer reviews) and could not be converted to pure CSS to leverage performance benefits of server side rendering. These are difficult areas to improve layout shifts because content (thus height) will vary. In those cases, we **wrapped the component in a container with a `min-height` applied**, predetermined based on observation of the average height for each of the specific components. The min-height is removed once the inner dynamic component is done rendering. While not perfect, this solution allowed us to reduce layout shift by a lot.
```

Comment:
```suggestion
Certain components on our customers sites have dynamic content (for example, a list of external customer reviews) and could not be converted to pure CSS to leverage performance benefits of server side rendering. These are difficult areas to improve layout shifts because content (thus height) will vary. In those cases, we wrapped the component in a container with a `min-height` applied, predetermined based on observation of the average height for each of the specific components. The `min-height` is removed once the inner dynamic component is done rendering. While not perfect, this solution allowed us to reduce layout shift by a lot.
```